src/encoder/colortype.rs

```diff
@@ -0,0 +1,72 @@
+/// Trait for different colortypes that can be encoded.
+pub trait ColorType {
```

Comment:
So, regarding how you weren't sure that this was a good idea. On one hand here's a bit of overlap between this and the `ColorType` enum at the crate root, but on the other hand having a trait with this `Inner` associated type is convenient because it allows functions like `write_image<C: ColorType>(&mut self, width: u32, height: u32, data: &[C::Inner]) -> TiffResult<()>` instead of having functions for every possible subpixel type. I would probably have ended up with something similar. In any case, I don't have a better idea right now, I'd say it's ok.

Reply:
The only other thing that I could think of, would be to make `write_image` like this:
```rust
fn write_image<T: TiffValue>(&mut self, width: u32, height: u32, colortype: ColorType, data: T) -> TiffResult<()>
```
However you then loose some type safety and the user becomes sort of locked to what's supported by `ColorType`, so I prefer this solution.